tests/PHPUnit/System/ComplicatedSegmentTest.php

```diff
@@ -44,8 +44,11 @@ public function getApiForTesting()
         $idSite = self::$idSite;
         $dateTime = self::$dateTime;
 
+        // Note: Creating a segment in the UI would generate double url encoded values
+        // But as segments might also be created through API, or provided as URL parameter, this might not always be the case
+        // Therefor some values are urlencoded twice, some aren't
```

Comment:
Should we have a combination of double url encoding and no encoding for all different combinations?

Reply:
We could have a lot more tests around segmentation. But the number of possible combinations is nearly unlimited.
Segments can fully be urlencoded or only the values can be encoded and that more even than once. In theory both could also be combined. The change here was more to check the special handling of backslashes, that can be used to escape a `,` or `;`, but can also be included in a segment as normal character. I wouldn't invest more time here for now and merge it. We can add more tests later anyway.